### PR TITLE
chore: Add Progress components

### LINKED ISSUE #161: Add Progress

### PR DESCRIPTION
**Pull request by soulcramer**
## 🧑‍ Changes description

Add Progress components

## 📸 Screenshots

Integrated in the review files

## 👋 Other info
close #161

### CODE REVIEW

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ * [ProgressIndicatorDefaults.ProgressAnimationSpec] as the default recommended [AnimationSpec] when
+ * animating progress
+ *
+ * @param progress the progress of this progress indicator, where 0.0 represents no progress and 1.0
+ * represents full progress. Values outside of this range are coerced into the range.
+ * @param modifier the [Modifier] to be applied to this progress indicator
+ */
+@Composable
+fun CircularProgressIndicator(
+    progress: Float,
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:84]: I think we should provide a FloatRange to help devs catch a potential error as soon as possible.

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                .size(CircularSize),
+            progress = progress,
+            color = color,
+        )
+    }
+}
+
+/**
+ * <a href="https://github.mpi-internal.com/leboncoin/android-app/blob/delivery/libraries/Compose/ProgressIndicators.md" class="external" target="_blank">Determinate Spark linear progress indicator</a>.
+ *
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:71]: 🟥 -2: Let's not use links to private docs (there are other occurrences in this PR)

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    } else {
+        MaterialCircularProgressIndicator(
+            modifier = modifier
+                .sparkUsageOverlay()
+                .size(CircularSize),
+            progress = progress,
+            color = color,
+        )
+    }
+}
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:67]: I feel like this comment is not useful

```suggestion
    if (isIndeterminate) {
        MaterialCircularProgressIndicator(
            modifier = modifier
                .sparkUsageOverlay()
                .size(CircularSize),
            color = ProgressIndicatorDefaults.circularColor,
        )
    } else {
        MaterialCircularProgressIndicator(
            modifier = modifier
                .sparkUsageOverlay()
                .size(CircularSize),
            progress = progress,
            color = ProgressIndicatorDefaults.circularColor,
        )
    }
}
```

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+/**
+ * <a href="https://github.mpi-internal.com/leboncoin/android-app/blob/delivery/libraries/Compose/ProgressIndicators.md" class="external" target="_blank">Determinate Spark linear progress indicator</a>.
+ *
+ * Progress indicators express an unspecified wait time or display the duration of a process.
+ *
+ * By default there is no animation between [progress] values. You can use
+ * [ProgressIndicatorDefaults.ProgressAnimationSpec] as the default recommended [AnimationSpec] when
+ * animating progress
+ *
+ * @param progress the progress of this progress indicator, where 0.0 represents no progress and 1.0
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:78]: I don't think this comment is legit. We are using `check(progress in 0.0f..1.0f)` that will not do coersion, but throw if the value is outside the expected range.

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/LinearProgressIndicator.kt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            trackColor = trackColor,
+        )
+    } else {
+        MaterialLinearProgressIndicator(
+            modifier = modifier.sparkUsageOverlay(),
+            progress = progress,
+            color = color,
+            trackColor = trackColor,
+        )
+    }
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/LinearProgressIndicator.kt:63]: Same here?

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/LinearProgressIndicator.kt [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+ * [Determinate Spark linear progress indicator](https://github.mpi-internal.com/leboncoin/android-app/blob/delivery/libraries/Compose/ProgressIndicators.md)
+ *
+ * Progress indicators express an unspecified wait time or display the duration of a process.
+ *
+ * By default there is no animation between [progress] values. You can use
+ * [ProgressIndicatorDefaults.ProgressAnimationSpec] as the default recommended [AnimationSpec] when
+ * animating progress
+ *
+ * @param progress the progress of this progress indicator, where 0.0 represents no progress and 1.0
+ * represents full progress. Values outside of this range are coerced into the range.
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/LinearProgressIndicator.kt:77]: same

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt (hunk 1 above)
Comment (soulcramer) [reply spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:84]: Yep I agree 👍

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt (hunk 4 above)
Comment (soulcramer) [reply spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt:77]: I removed the check an used a `coerceIn` instead

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt (hunk 2 above)
Comment (soulcramer) [reply spark/src/main/kotlin/com/adevinta/spark/components/progress/CircularProgressIndicator.kt]: They're removed

File: spark/src/main/kotlin/com/adevinta/spark/tokens/PaletteTokens.kt [hunk 7]
```diff
@@ -177,7 +177,7 @@ internal object PaletteTokens {
     internal val Surfer60 = Color(0xFF9C9BA4)
     internal val Surfer70 = Color(0xFFB6B5BC)
     internal val Surfer80 = Color(0xFFD1D0D5)
-    internal val Surfer90 = Color(0xFF7B7986)
+    internal val Surfer90 = Color(0xFFEBEBED)
```
Comment (SimonMarquis) [inline spark/src/main/kotlin/com/adevinta/spark/tokens/PaletteTokens.kt:180]: Hopefully this is an expected change 🤞

**Review by SimonMarquis — DISMISSED** [review]

**Review by SimonMarquis — APPROVED** [review]

**Pull request merged**